src/core/ext/lb_policy/round_robin/round_robin.c

```diff
@@ -251,6 +280,8 @@ static void rr_destroy(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol) {
     gpr_free(elem);
     elem = tmp;
   }
+
+  gpr_free(p->user_data);
```

Comment:
Ah! The misleading naming of variables... the RR policy *does* own that, which is not the user_data itself but an array of pointers to hold the actual not-owned user_data pointers. I've renamed the variable.